src/prof_recent.c

```diff
@@ -556,5 +642,18 @@ prof_recent_init() {
 	}
 	prof_recent_alloc_dummy->next = prof_recent_alloc_dummy;
 
+	assert(prof_recent_alloc_dummy_reserve == NULL);
```

Comment:
Allocate the two nodes together is fine.

Reply:
You mean to allocate a size-2 array?

Reply:
Right. That way I assume we can simply switch between the two items?

Reply:
It won't be easier to switch. There's still the same logic like:
```
prof_recent_t *dump_dummy = prof_recent_alloc_dummy;
prof_recent_alloc_dummy = prof_recent_alloc_dummy_reserve;
/* ... dump the dump_dummy list and then plug it back to the prof_recent_alloc_dummy list ... */
prof_recent_alloc_dummy_reserve = dump_dummy;
```
Any idea how it can be simplified?

Putting the two pointers in an array does simplify the initialization logic though, which can be a solid enough supporting argument itself.

Reply:
Yeah we would still need to do the swapping, we can only maintain a single global pointer of the array though (i.e. no array[0] is always the recent records, while array[1] always points to the dumping one).

Reply:
Right, but even in that case, I would macro define 0 and 1, so that it looks like `prof_recent_alloc_dummy[REAL]` and `prof_recent_alloc_dummy[RESERVE]`, rather than barebone `prof_recent_alloc_dummy[0]` and `prof_recent_alloc_dummy[1]`. It would end up looking pretty much the same if I just define two named pointers each pointing to one position in the array. Thoughts?

Reply:
Yeah it won't be simplified too much, the thing I want from it is to have only one global variable. We should also add a getter to wrap around the array.

Reply:
Realized that we cannot define an array for the two dummies, because they'll be individually dynamic: `prof_recent_alloc_dummy` can be shifted during prof allocation, and may also be deleted in `mallctl`. The same is true for `prof_recent_alloc_dummy_reserve`, when it's swapped to be used as `prof_recent_alloc_dummy`.

Extracted out the swapping operation into a utility function. Also avoided passing `dump_dummy` around, since it's always `prof_recent_alloc_dummy_reserve` (if the swap is fully completed in `prof_recent_alloc_dump_prepare_locked()` rather than having half of it left to `prof_recent_alloc_dump_finish_locked()`.